src/daemon.rs

```diff
@@ -129,8 +132,13 @@ struct Connection {
 
 fn tcp_connect(addr: SocketAddr, signal: &Waiter) -> Result<TcpStream> {
     loop {
-        match TcpStream::connect(addr) {
-            Ok(conn) => return Ok(conn),
+        match TcpStream::connect_timeout(&addr, DAEMON_TIMEOUT) {
+            Ok(conn) => {
+                // can only fail if DAEMON_TIMEOUT is 0
+                conn.set_read_timeout(Some(DAEMON_TIMEOUT)).unwrap();
```

Comment:
we need to have separate timeouts for connection, read, and write.
Connections should be quick, but read and writes may be slow --- for example, a large batch request is made from electrs, and the chain daemon takes many seconds to respond.